datarepo-clienttests/src/main/java/deploymentscripts/ModularHelmChart.java

```diff
@@ -0,0 +1,348 @@
+   * delete in order for the databases (Stairway and Data Repo) to be wiped because of how they
+   * decide which pod will do the database migration.
+   */
+  public void waitForDeployToFinish() throws Exception {
+    int pollCtr = Math.floorDiv(maximumSecondsToWaitForDeploy, secondsIntervalToPollForDeploy);
+
+    // first wait for the datarepo-api deployment to report "deployed" by helm ls
+    System.out.println("Checking Helm status of datarepo-api deployment");
+    boolean foundHelmStatusDeployed = false;
+    while (pollCtr >= 0) {
+      // list the available deployments
+      // e.g. helm ls --namespace mm
+      ArrayList<String> listCmdArgs = new ArrayList<>();
+      listCmdArgs.add("ls");
+      listCmdArgs.add("--namespace");
+      listCmdArgs.add(serverSpecification.namespace);
+      Process helmListProc = ProcessUtils.executeCommand("helm", listCmdArgs);
+      List<String> cmdOutputLines = ProcessUtils.waitForTerminateAndReadStdout(helmListProc);
+      for (String cmdOutputLine : cmdOutputLines) {
+        System.out.println(cmdOutputLine);
+      }
+
+      for (String cmdOutputLine : cmdOutputLines) {
+        if (cmdOutputLine.startsWith(serverSpecification.namespace + "-jade-datarepo-api")) {
+          if (cmdOutputLine.contains("deployed")) {
+            foundHelmStatusDeployed = true;
+            break;
+          }
+        }
+      }
+
+      if (foundHelmStatusDeployed) {
+        break;
+      }
+
+      TimeUnit.SECONDS.sleep(secondsIntervalToPollForDeploy);
+      pollCtr--;
+    }
+
+    // then wait for the datarepo-api deployment to respond successfully to a status request
+    System.out.println("Checking service status endpoint");
+    ApiClient apiClient = new ApiClient();
+    apiClient.setBasePath(serverSpecification.uri);
+    UnauthenticatedApi unauthenticatedApi = new UnauthenticatedApi(apiClient);
+    while (pollCtr >= 0) {
+      // call the unauthenticated status endpoint
+      try {
+        unauthenticatedApi.serviceStatus();
+        int httpStatus = unauthenticatedApi.getApiClient().getStatusCode();
+        System.out.println("Service status: " + httpStatus);
+        if (HttpStatusCodes.isSuccess(httpStatus)) {
+          break;
+        }
+      } catch (ApiException apiEx) {
+        System.out.println("Exception caught while checking service status: " + apiEx.getMessage());
+      }
+
+      TimeUnit.SECONDS.sleep(secondsIntervalToPollForDeploy);
+      pollCtr--;
+    }
+  }
+
+  /**
+   * Loop through the original datarepo-api YAML file and modify or add environment variables. Write
+   * the result to the specified output file.
+   *
+   * @param inputFile the original datarepo-api YAML file
+   * @param outputFile the modified datarepo-api YAML file
+   */
+  private void parseAndModifyApiYamlFile(File inputFile, File outputFile) throws IOException {
+    // the file line-by-line parsing below can be brittle because the YAML format is strictly
+    // enforced
```

Comment:
Do you think the brittle-ness of this is something to worry about?

Reply:
You could use jackson to parse the yaml into java objects, make modifications, and use jackson to write it back out

Reply:
I didn't know Jackson handled YAML, but that would be great. Will look into it.

Regarding the brittleness, I think we're just at the mercy of our deployment YAML structure either way. Any updates there will need to be reflected here or at least considered to make sure they won't this parsing.